crates/accelerate/src/synthesis/evolution/mod.rs

```diff
@@ -0,0 +1,439 @@
+            hit_paulis[i] = true;
+        }
+    }
+
+    for gate in gates {
+        out_gates.push(qiskit_clifford_gate(gate));
+
+        cur_paulis.conjugate_with_gate(gate);
+
+        // check which paulis are hit now
+        for i in 0..cur_paulis.len() {
+            if !hit_paulis[i] && cur_paulis.support_size(i) == 1 {
+                out_gates.push(qiskit_rotation_gate(py, &cur_paulis, i, &angles[i]));
+                hit_paulis[i] = true;
+            }
+        }
+    }
+
+    (out_gates, global_phase)
+}
+
+/// Return a Qiskit circuit with Clifford gates and rotations.
+///
+/// The rotations are assumed to be ordered (up to commutativity).
+///
+/// # Arguments
+///
+/// * py: a GIL handle, needed to negate rotation parameters in Python space.
+/// * gates: the sequence of Rustiq's Clifford gates returned by Rustiq's
+///     pauli network synthesis algorithm.
+/// * paulis: Rustiq's data structure storing the pauli rotations.
+/// * angles: Qiskit's rotation angles corresponding to the pauli rotations.
+fn inject_rotations_ordered(
+    py: Python,
+    gates: &Vec<CliffordGate>,
+    paulis: &PauliSet,
+    angles: &[Param],
+) -> (Vec<QiskitGate>, Param) {
+    let mut out_gates: Vec<QiskitGate> = Vec::with_capacity(gates.len() + paulis.len());
+    let mut global_phase = Param::Float(0.0);
+
+    let mut cur_paulis = paulis.clone();
+    let mut hit_paulis: Vec<bool> = vec![false; cur_paulis.len()];
+
+    let mut dag = CommutativityDag::from_paulis(paulis);
+
+    // check which paulis are hit at the very start
+    for i in 0..cur_paulis.len() {
+        let pauli_support_size = cur_paulis.support_size(i);
+        if pauli_support_size == 0 {
+            // in case of an all-identity rotation, update global phase by subtracting
+            // the angle
+            global_phase = radd_param(global_phase, multiply_param(&angles[i], -1.0, py), py);
+            hit_paulis[i] = true;
+            dag.remove_node(i);
+        } else if pauli_support_size == 1 && dag.is_front_node(i) {
+            out_gates.push(qiskit_rotation_gate(py, &cur_paulis, i, &angles[i]));
+            hit_paulis[i] = true;
+            dag.remove_node(i);
+        }
+    }
+
+    for gate in gates {
+        out_gates.push(qiskit_clifford_gate(gate));
+
+        cur_paulis.conjugate_with_gate(gate);
+
+        // check which paulis are hit now
+        for i in 0..cur_paulis.len() {
+            if !hit_paulis[i] && cur_paulis.support_size(i) == 1 && dag.is_front_node(i) {
+                out_gates.push(qiskit_rotation_gate(py, &cur_paulis, i, &angles[i]));
+                hit_paulis[i] = true;
+                dag.remove_node(i);
+            }
+        }
+    }
+
+    (out_gates, global_phase)
+}
+
+/// Return the vector of Qiskit's gate corresponding to the given vector
+/// of Rustiq's Clifford gate.
+fn qiskit_clifford_gates(gates: &Vec<CliffordGate>) -> Vec<QiskitGate> {
```

Comment:
Just a suggestion, since this is a transformation (I sort of expected this to return all Qiskit Clifford gates) 
```suggestion
fn to_qiskit_clifford_gates(gates: &Vec<CliffordGate>) -> Vec<QiskitGate> {
```

Reply:
Done (the commit will soon follow)